tmt/steps/__init__.py

```diff
@@ -144,6 +144,12 @@ def wake(self):
                 f"Insert login plugin with order '{plugin.order}'.", level=2)
             self._plugins.append(plugin)
 
+        # Insert reboot plugins if requested on the command line
+        for plugin in Reboot.plugins(step=self):
```

Comment:
Just a quick note, I modified this part in https://github.com/psss/tmt/pull/833 to make login work properly when a step fails (there was also a slight change to parsing phases). I am not sure what effect it has on this PR, it may not have any effect at all :)